pulsar/crypto/crypto_failure_action.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package crypto
+
+type ProducerCryptoFailureAction int
+type ConsumerCryptoFailureAction int
+
+const (
```

Comment:
Since there aren't really enums in go to namespace things maybe we should use long names to make it easier to reason about in the code?

ProducerCryptoFailureActionFail or ProducerFailureActionSend
ProducerCryptoFailureActionSend or ProducerFailureActionFail

Thoughts?

Reply:
I tried to use them as enum, but yeah I think it make sense to have a long names instead.

Reply:
And it is good have long name like `ProducerCryptoFailureActionFail` instead of `ProducerFailureActionSend`